impeller/renderer/backend/gles/gpu_tracer_gles.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/renderer/backend/gles/gpu_tracer_gles.h"
+#include <thread>
+#include "fml/trace_event.h"
+
+namespace impeller {
+
+GPUTracerGLES::GPUTracerGLES(const ProcTableGLES& gl, bool enable_tracing) {
+#ifdef IMPELLER_DEBUG
+  auto desc = gl.GetDescription();
+  enabled_ =
+      enable_tracing && desc->HasExtension("GL_EXT_disjoint_timer_query");
+#endif  // IMPELLER_DEBUG
+}
+
+void GPUTracerGLES::MarkFrameStart(const ProcTableGLES& gl) {
+  if (!enabled_ || active_frame_.has_value() ||
+      std::this_thread::get_id() != raster_thread_) {
+    return;
+  }
+
+  // At the beginning of a frame, check the status of all pending
+  // previous queries.
+  ProcessQueries(gl);
+
+  uint32_t query = 0;
+  gl.GenQueriesEXT(1, &query);
+  if (query == 0) {
+    return;
+  }
+
+  FML_DCHECK(!active_frame_.has_value());
```

Comment:
This duplicates the `active_frame_.has_value()` check above

Reply:
Done